dist/test_tag.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+##### LOCAL VARIABLES AND CONSTANTS #####
+component = ""
+testingArea = ""
+testType = ""
+
+isParsingTag = False
+
+nbTests = 0
+
+sortedTags = []
+testFiles = []
+
+taggedFiles = {}
+validTags = {}
+
+END_TAG = "[END_TAGS]"
+NB_TAG_ARGS = 3
+START_TAG = "[TEST_TAGS]"
+#####
+
+##### GET ALL TEST FILES #####
+for root, dirs, files in os.walk("../test/"):
+    path = root.split(os.sep)
+    for file in files:
+        filename = os.path.join('/'.join(path), file)
+        if filename.endswith("main.c") or filename.endswith(".py"):
+            testFiles.append(filename)
+#####
+
+##### RETRIEVE VALID TAGS #####
+validationFile = open("test_tags.ok", "r")
+
+# The file has the following pattern
+# <COMPONENT>:<TESTING_TYPE>:<TESTING_AREA>:<DESCRIPTION>
+# A tag is made of the three first values: COMPONENT, TEST_TYPE and TESTING_AREA
+tags = validationFile.readlines()
+tags = [tag.replace('\n', '') for tag in tags]
+
+for tag in tags:
+    currentLine = tag.split(':')
+    # Createa key value pair <TAG>:<DESCRIPTION>
+    validTags[':'.join(currentLine[:NB_TAG_ARGS])] = ':'.join(currentLine[NB_TAG_ARGS:])
+
+validationFile.close()
+#####
+
+##### PARSE TEST FILES #####
+for filename in testFiles:
+    inputFile = open(filename, "r")
+
+    lines = inputFile.readlines()
+
+    # Read line by line
+    for line in lines:
+
+        # Format line
+        line = line.replace('\n', '').replace('\r', '') \
+                   .replace(' ', '').replace('#', '') \
+                   .replace('*', '')
+
```

Comment:
Can we add some functionality (which will be turned off initially by a flag) that outputs any test file name that isn't tagged? We'll also need to add an exclusion tag for files that match our test filter but aren't actual test files. e.g. "test/suite/wttest.py". I think in that scenario the can have a dead tag similar to:
```
# [TEST_TAGS]
# ignored_file
# [END_TAGS]
```

The intention of this functionality is that we can switch it on once this project has completed and we will force people to use our tagging system. Whether we end up turning it on is another question, but there isn't much point implementing a test tagging system if we don't end up forcing people to use it.

Reply:
See line 27 for IGNORE_FILE tag
See line 42 for -p option
Added -i option for stats line 40